lib/matplotlib/pyplot.py

```diff
@@ -230,24 +230,51 @@ def switch_backend(newbackend):
     _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
 
 
-def show(*args, **kw):
+def show(block=None):
     """
-    Display a figure.
-    When running in ipython with its pylab mode, display all
-    figures and return to the ipython prompt.
+    Draw and display all figures.
 
-    In non-interactive mode, display all figures and block until
-    the figures have been closed; in interactive mode it has no
-    effect unless figures were created prior to a change from
-    non-interactive to interactive mode (not recommended).  In
-    that case it displays the figures but does not block.
+    Mainly used in non-interactive mode to draw and display
+    the figures. It usually does nothing in interactive mode.
```

Comment:
Why would it not do anything in interactive mode? I think it also shows the figure there.

Reply:
I am not sure. It was written in the old docstring that it does nothing. 

When I test it in spyder, the only difference is that the "thumbnail" in the toolbar start to blink if plt.show() is used. The figure window don't pop up to the front or something like that.